code/modules/admin/verbs/map_export.dm

```diff
@@ -0,0 +1,261 @@
+				//If there is nothing there, save as a noop (For odd shapes)
+				if(!place)
+					place = /turf/template_noop
+					location = /area/template_noop
+					objects = list()
+				//Ignore things in space, must be a space turf
+				else if(istype(place, /turf/open/space) && !(save_flag & SAVE_SPACE))
+					place = /turf/template_noop
+					location = /area/template_noop
+				//Stuff to add
+				else
+					location = place_area.type
+					objects = place
+					place = place.type
+				//====Saving shuttles only / non shuttles only====
+				var/is_shuttle_area = istype(location, /area/shuttle)
+				if((is_shuttle_area && shuttle_area_flag == SAVE_SHUTTLEAREA_IGNORE) || (!is_shuttle_area && shuttle_area_flag == SAVE_SHUTTLEAREA_ONLY))
+					place = /turf/template_noop
+					location = /area/template_noop
+					objects = list()
+				//====For toggling not saving areas and turfs====
+				if(!(save_flag & SAVE_AREAS))
+					location = /area/template_noop
+				if(!(save_flag & SAVE_TURFS))
+					place = /turf/template_noop
+				//====Generate Header Character====
+				var/header_char = calculate_tgm_header_index(index, layers)	//The characters of the header
+				var/current_header = "(\n"										//The actual stuff inside the header
+				//Add objects to the header file
+				var/empty = TRUE
+				//====SAVING OBJECTS====
+				if(save_flag & SAVE_OBJECTS)
+					for(var/obj/thing in objects)
+						CHECK_TICK
+						if(thing.type in obj_blacklist)
+							continue
+						var/metadata = generate_tgm_metadata(thing)
+						current_header += "[empty ? "" : ",\n"][thing.type][metadata]"
+						empty = FALSE
+						//====SAVING SPECIAL DATA====
+						//This is what causes lockers and machines to save stuff inside of them
+						if(save_flag & SAVE_OBJECT_PROPERTIES)
+							var/custom_data = thing.on_object_saved()
+							current_header += "[custom_data ? ",\n[custom_data]" : ""]"
+				//====SAVING MOBS====
+				if(save_flag & SAVE_MOBS)
+					for(var/mob/living/thing in objects)
+						CHECK_TICK
+						if(istype(thing, /mob/living/carbon))		//Ignore people, but not animals
+							continue
+						var/metadata = generate_tgm_metadata(thing)
+						current_header += "[empty ? "" : ",\n"][thing.type][metadata]"
+						empty = FALSE
+				current_header += "[empty ? "" : ",\n"][place],\n[location])\n"
+				//====Fill the contents file====
+				//Compression is done here
+				var/position_of_header = header_dat.Find(current_header)
+				if(position_of_header)
+					//If the header has already been saved, change the character to the other saved header
+					header_char = header_chars[position_of_header]
+				else
+					header += "\"[header_char]\" = [current_header]"
+					header_chars += header_char
+					header_dat += current_header
+					index ++
+				contents += "[header_char]\n"
+			contents += "\"}"
+	return "//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE\n[header][contents]"
```

Comment:
turn `//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE` into a `#define` macro like `#define DMM2TGM_MESSAGE ...` in order to help us against drift.